scalafix-cli/src/main/scala/scalafix/internal/v1/Args.scala

```diff
@@ -370,11 +374,48 @@ case class Args(
       } else {
         classpath
       }
-    Classpath(targetrootClasspath) ++ baseClasspath
+    val finalClassPath = Classpath(targetrootClasspath) ++ baseClasspath
+    val badEntries = finalClassPath.entries.filterNot {
+      ensureValidClasspathEntry
+    }
+    if (badEntries.isEmpty) {
+
+      Configured.ok(finalClassPath)
+    } else {
+      val extensions = validClasspathFileExtension.mkString("[", ",", "]")
+      val prefix = s"""|Invalid classpath entry. Make sure your classpath
+        |contains only directories or file that ends with
+        |the following extension: ${extensions}. The following path are invalid: """.stripMargin
+      val error =
+        badEntries.map(_.toString() + "\n").mkString(prefix, " - ", "")
+      Configured.notOk(ConfError.message(error))
+    }
+  }
+
+  private val validClasspathFileExtension = List("jar", "zip", "semanticdb")
+
+  /**
+   * Rules are as follow:
+   *   - valid if it is a directory
+   *   - valid if it is a file and it ends with `jar`, `zip` or `semanticdb`
+   *
+   * We also allow path that don't exists: some tests uses dummy data, but its
+   * also possible that projects out there have this already do this and are not
+   * broken today.
+   */
+  private def ensureValidClasspathEntry(path: AbsolutePath) = {
+    !Files.exists(path.toNIO) ||
+    path.isDirectory || {
+      val fileName = path.toNIO.getFileName()
+      lazy val hasRightExtension = validClasspathFileExtension.exists(ext =>
+        fileName.toString.endsWith(s".$ext")
+      )
```

Comment:
yeah, that makes sense, so if there is an extension, I validated it, otherwise, I let it go through. I could add some error handling to catch that nasty zip header error to be more helpful

Reply:
> if there is an extension, I validate it, otherwise, I let it go through. I could add some error handling to catch that nasty zip header error to be more helpful

Agreed, my suggestion is to open the file with `JarFile` / `ZipFile` when the extension is not .jar or .zip. If that fails, then we raise a validation error since scalameta will fail later on. This strategy would remove the false positives without introducing too much overhead.

In any case (with the current state of the PR or with my suggestion), false negative would remain, as invalid .jar or .zip would still cause scalameta to fail later on (while the JRE would just ignore them), but that seems like a corner case. As I mentioned in the overall comment, the only way to address that without introducing an IO overhead parsing all files as ZIP in scalafix would be to change the scalameta behavior to ignore these entries.

Reply:
Do you mind if I leave this PR as is and go upstream to scala meta and address that?

Reply:
That would be great, aligning scalameta with the JRE behavior would be the best (apologies for not driving the discussion towards that from the beginning). If this can't be done, then let's resume discussions here.